backend/src/Edna/Config/Environment.hs

```diff
@@ -0,0 +1,86 @@
+module Edna.Config.Environment
+  ( EdnaEnvironments(..)
+
+  , ednaEnvs
+  , askDebugDB
+  ) where
+
+import Prelude (read)
+import Universum
+
+import Data.ByteString.Char8 (pack)
+import qualified Data.Char as C
+import Edna.Config.Definition (LoggingConfig(..))
+import Edna.Util (ConnString(..), NetworkAddress)
+import System.Environment (lookupEnv)
+
+-- If @EDNA_DB_DEBUG@ environment variable is set to @1@
+-- or @TRUE@ (case-insensitive), we will run postgres with
+-- debug logging.
+askDebugDB :: IO Bool
+askDebugDB = lookupEnv "EDNA_DB_DEBUG" <&> \case
+  Just "1"                       -> True
+  Just (map C.toLower -> "true") -> True
+  _                              -> False
+
+data EdnaEnvironments = EdnaEnvironments
+  { eeListen :: IO (Maybe NetworkAddress)
+  , eeDocs :: IO (Maybe Bool)
+  , eeConnection :: IO (Maybe ConnString)
+  , eeMaxCon :: IO (Maybe Int)
+  , eeLogging :: IO (Maybe LoggingConfig)
+  } deriving stock (Generic)
+
+ednaEnvs :: EdnaEnvironments
+ednaEnvs = EdnaEnvironments
+  { eeListen = apiListenEnv
+  , eeDocs = serveDocsEnv
+  , eeConnection = connectionEnv
+  , eeMaxCon = dbMaxConnectionsEnv
+  , eeLogging = loggingLevelEnv
+  }
+
+apiListenEnv :: IO (Maybe NetworkAddress)
+apiListenEnv = do
+  env <- lookupEnv "EDNA_API_LISTEN"
+  return $ read <$> env
+
+serveDocsEnv :: IO (Maybe Bool)
+serveDocsEnv = lookupEnv "EDNA_SERVE_DOCS" <&> \case
+  Just "1"                       -> Just True
+  Just (map C.toLower -> "true") -> Just True
+  _                              -> Nothing
```

Comment:
It duplicates `askDebugDB`, can you please extract this logic into a separate helper function?

Reply:
sure